File: app/src/main/java/rusha/x/product/ProductDetailsViewModel.java
package rusha.x.product;

import androidx.lifecycle.MutableLiveData;

import javax.inject.Inject;

import rusha.x.BaseViewModel;

public class ProductDetailsViewModel extends BaseViewModel {

    @Inject
    public ProductDetailsViewModel() {
        super();
    }

    public MutableLiveData<String> nameLiveData = new MutableLiveData<>("");
    public MutableLiveData<String> formattedPriceLiveData = new MutableLiveData<>("");

    public MutableLiveData<ShowWelcomeParams> showWelcomeLiveData = new MutableLiveData<>(null);

    public void init(Product product) {
        nameLiveData.setValue(product.name);
        formattedPriceLiveData.setValue(product.price + "/" + product.unit);
    }

    public void onResume() {
        showWelcomeLiveData.setValue(new ShowWelcomeParams("Привет!"));
        showWelcomeLiveData.setValue(null);
    }
}

File: app/src/main/java/rusha/x/product/ProductDetailsViewModel.kt
package rusha.x.product

import androidx.lifecycle.MutableLiveData
import androidx.lifecycle.ViewModel

class ProductDetailsViewModel : ViewModel() {

    val nameLiveData = MutableLiveData<String>("")
    val formattedPriceLiveData = MutableLiveData<String>("")

    data class ShowWelcomeParams(val text: String)

    val showWelcomeLiveData = MutableLiveData<ShowWelcomeParams?>(null)

    fun init(product: Product) {
        nameLiveData.value = product.name
        formattedPriceLiveData.value = "${product.price}/${product.unit}"
    }

    fun onResume() {
        showWelcomeLiveData.value = ShowWelcomeParams(text = "Привет!")
        showWelcomeLiveData.value = null
    }
}